crates/miden-objects/src/transaction/inputs.rs

```diff
@@ -531,3 +535,170 @@ pub fn validate_account_seed(
         (false, None) => Ok(()),
     }
 }
+
+// FOREIGN ACCOUNT INPUTS
+// ================================================================================================
+
+/// Contains information about a foreign account, with everything required to execute its code.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct ForeignAccountInputs {
+    /// Account header of the foreign account.
+    account_header: AccountHeader,
+    /// Header information about the account's storage.
+    storage_header: AccountStorageHeader,
+    /// Code associated with the account.
+    account_code: AccountCode,
+    /// Merkle proof of the account's inclusion in the account tree.
+    account_witness: MerklePath,
+    /// Storage SMT proof for storage map values that the transaction will access.
+    storage_map_proofs: Vec<SmtProof>,
+}
```

Comment:
Not for this PR, but I wonder if we should just have one general `AccountInputs`. It could look something like:
```rust
pub struct AccountInputs {
    account_header: AccountHeader,
    storage_header: AccountStorageHeader,
    code_header: AccountCodeHeader,
    witness: AccountWitness,
    storage_map_proofs: Vec<SmtProof>,
    vault_proofs: Vec<SmtProof>,
}

pub struct AccountCodeHeader {
    procedures: Vec<AccountProcedureInfo>,
}
```
This struct would contain the data needed to load the account into the transaction kernel, irrespective of whether this is a foreign or native account.

Let's create an issue for this.

Reply:
Created #1296